src/planner/Algo.h

```diff
@@ -92,6 +92,47 @@ class ProduceAllPaths final : public SingleInputNode {
     ProduceAllPaths(QueryContext* qctx, PlanNode* input)
         : SingleInputNode(qctx, Kind::kProduceAllPaths, input) {}
 };
+
+class CartesianProduct final : public SingleDependencyNode {
+public:
+    static CartesianProduct* make(QueryContext* qctx, PlanNode* input) {
+        return qctx->objPool()->add(new CartesianProduct(qctx, input));
+    }
+
+    Status addVar(std::string varName) {
```

Comment:
please move the implementation of this function to cpp file, don't put all codes into header file.

Reply:
done